Add Workbox detection

tomayac opened the pull request:
This PR adds detection of the increasingly popular [Workbox](https://developers.google.com/web/tools/workbox/) library that is used in Service Workers. 

Detection of Workbox is asynchronous and obviously depends on the presence of a Service Worker, which required some minor changes in the way detection works. Namely, I needed to turn the test function into an `async` function and `await` its results. This means all other existing functions can remain as they are, purely the logic needs a global `await`. Hope this can be merged.

FYI @jeffposnick.

tomayac commented on content_scripts/lib_detect.js:
```diff
@@ -36,15 +36,15 @@
-    window.setTimeout(function() {
-        wait();
+    window.setTimeout(async function() {
+        await wait();
```
😂 Love it!

tomayac commented:
Screenshot of this in action:

![screen shot 2018-09-21 at 13 56 43](https://user-images.githubusercontent.com/145676/45879848-41258980-bda6-11e8-974f-1ed6cc91db53.png)

jeffposnick commented:
I'm not sure how the versioning metadata is created, but Workbox tends to be loaded via an `importScripts()` with a URL that contains the version string. (E.g. `http://storage.googleapis.com/workbox-cdn/releases/3.6.1/workbox-sw.js` is v3.6.1) Could that be detected in this extension?

tomayac commented:
Per @jeffposnick's comment, added [semver](https://github.com/semver/semver/issues/232#issue-48635632)-based version number parsing in https://github.com/johnmichel/Library-Detector-for-Chrome/pull/121/commits/ab1288ffc0a8c437b9e2b5bc4fe5e4708ba39dc5. Updated screenshot:


![screen shot 2018-09-24 at 09 27 38](https://user-images.githubusercontent.com/145676/45940983-6d2a5000-bfdc-11e8-88b8-4ad4d4fbb541.png)

johnmichel commented:
LGTM.

Thanks!
